core/src/mindustry/core/Renderer.java

```diff
@@ -83,14 +85,15 @@ public Renderer(){
         Shaders.init();
 
         Events.on(ResetEvent.class, e -> {
-            shakeTime = shakeIntensity = 0f;
+            shakeTime = shakeIntensity = shakeReduction = 0f;
             camShakeOffset.setZero();
         });
     }
 
     public void shake(float intensity, float duration){
-        shakeIntensity = Math.max(shakeIntensity, intensity);
+        shakeIntensity = Math.max(shakeIntensity, Mathf.clamp(intensity, 0, 100));
         shakeTime = Math.max(shakeTime, duration);
+        shakeReduction = shakeIntensity / shakeTime;
```

Comment:
shouldn't this only update if the intensity and/or time updates?

Reply:
Since `update()` reduces both intensity and time, having `shakeIntensity / shakeTime` wouldn't reduce the amount accurately.
If I make the shake based on `shakeIntensity > 0` and then just not decrease shakeTime it would accurately decrease over time, however inputing a new shake time wouldn't work properly if the current time doesn't decrease.

The reduction rate needs to be calculated upon inputting shake parameters.